Mission1: Step3: 빼빼로 만들기

wolframhwang opened the pull request:
@llghdud921 제리
빼빼로 만드는 Pepero.Swift를 추가하였고, 그에 대한 플로우 차트를 반영하였습니다.
리뷰 부탁드리겠습니다~

wolframhwang committed:
[refactor] Mission1: Step2: 불필요한 주석 제거

불필요한 주석을 제거한 패치를 포함하고 있습니다.
코드를 읽는데 아무런 도움이 되지 않는 코드라 판단하여 제거합니다.

Signed-off-by: wolframhwang <hzw94@naver.com>

wolframhwang committed:
Revert "[refactor] Mission1: Step2: 불필요한 주석 제거"

This reverts commit fdf2f2ff915f0206bb9c868e0dcc3b3cd36cc9df.

wolframhwang committed:
[refactor] Mission1: Step2: 불필요한 주석 제거

코드 읽을 시 아무 역할을 하지 못하는 주석을 제거합니다.
함수 네이밍이 동사형으로 적절히 잘 적혀 있어, 그 외의 일반적 설명이 필요 없다고 판단했습니다.

Signed-off-by: wolframhwang <hzw94@naver.com>

wolframhwang committed:
[feat] Mission1: Step3: 빼빼로 출력 추가

빼빼로를 몸통 + 토핑 + 막대기를 출력할수 있는 코드를 작성한 패치를 추가합니다.

Signed-off-by: wolframhwang <hzw94@naver.com>

wolframhwang committed:
Revert "[feat] Mission1: Step3: 빼빼로 출력 추가"

This reverts commit e230ea0909b73ceac8651ac0dadf22fc047b42c6.

wolframhwang committed:
[feat] Mission1: Step3: 빼빼로 출력 추가

빼빼로를 몸통 + 토핑 + 막대기를 출력할수 있는 코드를 작성한 패치를 추가합니다.

Signed-off-by: wolframhwang <hzw94@naver.com>

wolframhwang committed:
[docs] Mission1: Step3: 빼빼로 만들기 Flow Chart 추가 반영

빼빼로 그리는 플로우 차트를 작성했습니다.

Signed-off-by: wolframhwang <hzw94@naver.com>

llghdud921 commented on CodeStarterCamp_Week1/Pepero.swift:
```diff
@@ -0,0 +1,52 @@
+//
+import Foundation
+
+func makePeperoBar(peperoHeight: Int) {
```
bar의 길이가 `peperoHeight`을 의미하는 것이 맞을까요?
함수명이 `peperoBar`를 나타내고 있어서 `height`가 더 명확할 것 같습니다 :)

llghdud921 commented on CodeStarterCamp_Week1/Pepero.swift:
```diff
@@ -0,0 +1,52 @@
+import Foundation
+
+func makePeperoBar(peperoHeight: Int) {
+    let rangeOfPeperoHeight = 0..<peperoHeight
```
여기도! 수정해보면 좋을 것 같아요!

llghdud921 commented on CodeStarterCamp_Week1/Pepero.swift:
```diff
@@ -0,0 +1,52 @@
+}
+
+func makePeperoBody(peperoBody: String, peperoToping: String, peperoBodyHeight: Int) {
+    let rangeOfPeperoHeight = 0..<peperoBodyHeight
```
`peperoBodyheight`를 이용한 범위의 변수명이 `rangeOfPeperoHeight`는 잘 맞지 않는 것 같습니다!
개선해볼 수 있을까요?

llghdud921 commented on CodeStarterCamp_Week1/Pepero.swift:
```diff
@@ -0,0 +1,52 @@
+}
+
+func makePeperoBodyLayer(peperoBody: String, peperoToping: String, peperoHeight: Int) {
+    if peperoHeight % 2 == 0 {
```
우선 해당코드를 실행했을 때 실행예시와 달랐습니다!
그 이유는 해당 조건문때문인 것 같은데요! 다시 체크바랍니다 ☺️
추가로 `peproHeight`도 필요하지 않은 것 같아요~

llghdud921 commented on CodeStarterCamp_Week1/Pepero.swift:
```diff
@@ -0,0 +1,52 @@
+func makePeperoBodyLayer(peperoBody: String, peperoToping: String, peperoHeight: Int) {
+    if peperoHeight % 2 == 0 {
+        print("\(peperoToping + peperoBody)")
+    }else {
```
```suggestion
    } else {
```
else앞뒤로 공백을 넣어주는 코딩 컨벤션이 있습니다!

wolframhwang replied:
✅

llghdud921 commented on CodeStarterCamp_Week1/Pepero.swift:
```diff
@@ -0,0 +1,52 @@
+
+
+
+func makePepero(peperoBodyLength: Int, peperoBodyContent: String, pepeproBodyTopingContent: String, peperoBarLength: Int) {
```
매개변수 네이밍을 정~~~~~~말 잘해주셨어요!!! 👍 
한가지 말씀드릴 점은 `Pepero`가 함수명에 포함되어 있어서 `bodyLength`, `bodyContent`, `BodyToppingContent`.. 이런 네이밍도 좋을 것 같아요 😊

llghdud921 commented on CodeStarterCamp_Week1/Pepero.swift:
```diff
@@ -0,0 +1,52 @@
+    print("토핑: \(pepeproBodyTopingContent)")
+    print("막대길이: \(peperoBarLength)")
+    
+    makePeperoBody(peperoBody: peperoBodyContent, peperoToping: pepeproBodyTopingContent, peperoBodyHeight: peperoBodyLength)
```
완벽한 함수 분리 💯

llghdud921 commented on CodeStarterCamp_Week1/Pepero.swift:
```diff
@@ -0,0 +1,52 @@
+    print()
+}
+
+makePepero(peperoBodyLength: 10, peperoBodyContent: "***", pepeproBodyTopingContent: " ", peperoBarLength: 4)
```
크~게 중요한 건 아니지만 Topping에 빈 공백 문자열을 넣어주셨는데 
Step3 요구사항에 나와있는 실행예시를 보시면 아래와 같이 써있어요!
> (일반 빼빼로의 좌측에는 빈칸이 하나 추가되어있습니다.)

```suggestion
makePepero(peperoBodyLength: 10, peperoBodyContent: " ***", pepeproBodyTopingContent: "", peperoBarLength: 4)
```
아마도 요것이지 아닐까 싶습니다ㅎㅎ

wolframhwang replied:
만약 이부분이 파라미터로 " ***" 같이 처리 된다면 일관성있는 케이스로 보여지진않아서
함수 내부에서 처리 하는걸로 바꾸겠습니다.

llghdud921 commented on CodeStarterCamp_Week1/Pepero.swift:
```diff
@@ -0,0 +1,52 @@
+    print("몸통: \(peperoBodyContent)")
+    print("토핑: \(pepeproBodyTopingContent)")
+    print("막대길이: \(peperoBarLength)")
+    
```
여기에 `print()`가 들어가야하지 않을까요...? 요구사항에 집착하는 사람이라... 🤣

llghdud921 reviewed (changes requested):
안녕하세요!!! 제리입니당
역시나 Step3도 너무 잘해주셨네요.. 👍 

### Good 
- 네이밍을 정말 잘하시네요!!! 전 아직도 어렵답니다....
- FlowChart도 너무 잘 만들어주셨어요 👍 

### 조언 드릴 부분
- 실행예시와 실제 실행했을 때 출력되는 부분이 달라서 당황스러웠어요 😅  제출하기 전 꼼꼼하게 체크해보셨으면 합니다!!

wolframhwang committed:
[refactor] Mission1: Step3: 빼빼로 만들기 수정사항 반영

1. 함수, 변수 네이밍을 변경했습니다.
2. 파라미터는 함수에 대상이 포함되어 있다면 내용물에 대해서만 지침하도록 네이밍
변경했습니다.
3. 빼빼로 높이에 따라 body layer가 변경 되는 것 같다고 생각해서 height 를 파라미터로 받아온 것이고, 요구사항에서는

몸통(String), 토핑(String)을 매개변수로 받아 토핑과 몸통을 그리는 함수
길이(Int)에 따라 몸통을 그리는 함수

이렇게 나와있었는데, 저런 요소 "만" 을가지고 만들라는 말이 없어서 임의로 추가 하긴 했었습니다.

4. 토핑 공백 관련도 말씀 주셨는데, 이건 제가 드래그 해보지 않은 잘못이 맞네요. 파리미터 공백으로 줘도 처리할수 있게 만들었습니다.
5. else 관련 코딩컨벤션 추가해줬습니다. SwiftLint 같이 컨벤션 라이브러리 써서 맞춰도 괜찮을것 같다는 생각이 들었습니다

Signed-off-by: wolframhwang <hzw94@naver.com>

wolframhwang committed:
[docs] Mission1: Step3: 빼빼로 만들기 플로우 차트 반영

빼빼로 만들기 로직 변경에 따른 플로우 차트 변경사항을 반영한 패치입니다.

Signed-off-by: wolframhwang <hzw94@naver.com>

llghdud921 commented on CodeStarterCamp_Week1/Pepero.swift:
```diff
@@ -0,0 +1,64 @@
+    }
+}
+
+func makePeperoBodyLayer(bodyContent: String, topingContent: String) -> (evenLayer: String, oddLayer: String) {
```
파라미터 레이블을 넣으시다니 👍 👍

llghdud921 reviewed (approved):
안녕하세요 제리에용
제가 실행결과가 다르다고 말씀을 드렸는데 알고보니 야곰에서 저희에게 주신 예시랑 캠퍼 요구사항과 다르다는 점을 알게되었어요...! 죄송합니다 😭

리팩토링 매우 잘해주셔서 머지 하도록 하겠습니다! 고생하셨어요~!!
내일부터 2주차인데 지금까지 잘해주신것처럼 화이팅 해보아여~
